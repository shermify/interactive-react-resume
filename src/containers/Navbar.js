import React from 'react';
import Menu from '../components/Menu';

class Navbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = { collapsed: true, offset: -50 };
  }
  handleToggle = (e) => {
    const sender = e.target.className;

    if (sender !== 'nav-link' || !this.state.collapsed) {
      const collapsed = !this.state.collapsed;
      const offset = collapsed ? -50 : -300;
      this.setState({ collapsed, offset });
    }
  }
  render() {
    return (<Menu
      toggle={this.handleToggle}
      collapsed={this.state.collapsed}
      offset={this.state.offset}
    />);
  }
}

export default Navbar;
