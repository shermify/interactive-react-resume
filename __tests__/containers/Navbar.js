import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Navbar from '../../src/containers/Navbar';

describe('<Navbar />', () => {
  const wrapper = mount(<Navbar />);

  it('renders correctly', () => {
    const tree = renderer.create(
      <Navbar />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should remain collapsed if already collapsed and sender is navlink', () => {
    const mockedEvent = { target: {
      className: 'nav-link',
    } };
    wrapper.instance().handleToggle(mockedEvent);
    expect(wrapper.state().collapsed).toEqual(true);
    expect(wrapper.state().offset).toEqual(-50);
  });

  it('should open if collapsed and sender is button', () => {
    const mockedEvent = { target: {
      className: 'button',
    } };
    wrapper.instance().handleToggle(mockedEvent);
    expect(wrapper.state().collapsed).toEqual(false);
    expect(wrapper.state().offset).toEqual(-300);
  });

  it('should close if open and sender is nav-link', () => {
    const mockedEvent = { target: {
      className: 'nav-link',
    } };
    wrapper.instance().handleToggle(mockedEvent);
    expect(wrapper.state().collapsed).toEqual(true);
    expect(wrapper.state().offset).toEqual(-50);
  });

  it('should close if open and sender is button', () => {
    const mockedEvent = { target: {
      className: 'button',
    } };
    wrapper.setState({ collapsed: false, offset: -300 });
    wrapper.instance().handleToggle(mockedEvent);
    expect(wrapper.state().collapsed).toEqual(true);
    expect(wrapper.state().offset).toEqual(-50);
  });
});
