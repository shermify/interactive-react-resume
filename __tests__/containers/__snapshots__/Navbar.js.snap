// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<div
  className="menu"
>
  <div
    className="animate-nav"
    style={
      Object {
        "maxHeight": "55px",
      }
    }
  >
    <nav
      className="navbar navbar-expand-lg"
    >
      <button
        className="navbar-toggler mb-3"
        onClick={[Function]}
        type="button"
      >
        <test-file-stub
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="navbar-collapse collapse show"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Profile
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Experience
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Skills
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
